src/components/Planning/ProjectProgrammedDialog/ProjectProgrammedDialog.test.tsx

```diff
@@ -0,0 +1,189 @@
+import { Route } from 'react-router';
+import PlanningView from '@/views/PlanningView/PlanningView';
+import { CustomContextMenu } from '@/components/CustomContextMenu';
+import { mockGroups } from '@/mocks/mockGroups';
+import { mockLocations, mockDistricts, mockDivisions } from '@/mocks/mockLocations';
+import { mockGetResponseProvider } from '@/utils/mockGetResponseProvider';
+import { addProjectUpdateEventListener, removeProjectUpdateEventListener } from '@/utils/events';
+jest.mock('axios');
+jest.mock('react-i18next', () => mockI18next());
+
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+const store = setupStore();
+
+const navigateToProjectRows = async (renderResult: CustomRenderResult) => {
+  const { user, store, getByTestId } = renderResult;
+  const { masterClasses, classes } = store.getState().class;
+  await user.click(getByTestId(`expand-${masterClasses[0].id}`));
+  await user.click(getByTestId(`expand-${classes[0].id}`));
+};
+const render = async () =>
+  await act(async () =>
+    renderWithProviders(
+      <Route
+        path="/"
+        element={
+          <>
+            <PlanningView />
+            <CustomContextMenu />
+          </>
+        }
+      >
+        <Route path=":masterClassId" element={<PlanningView />}>
+          <Route path=":classId" element={<PlanningView />}>
+            <Route path=":subClassId" element={<PlanningView />}>
+              <Route path=":districtId" element={<PlanningView />} />
+            </Route>
+          </Route>
+        </Route>
+      </Route>,
+      {
+        preloadedState: {
+          class: {
+            ...store.getState().class,
+            allClasses: mockProjectClasses.data,
+            masterClasses: mockMasterClasses.data,
+            classes: mockClasses.data,
+            subClasses: mockSubClasses.data,
+          },
+          location: {
+            ...store.getState().location,
+            allLocations: mockLocations.data,
+            districts: mockDistricts.data,
+            divisions: mockDivisions.data,
+          },
+          group: {
+            ...store.getState().group,
+            groups: mockGroups.data,
+          },
+          lists: {
+            ...store.getState().lists,
+            phases: mockProjectPhases.data,
+          },
+        },
+      },
+    ),
+  );
+describe('ProjectProgrammedDialog', () => {
+  beforeEach(() => {
+    mockGetResponseProvider();
+  });
+
+  afterEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it('renders the component wrappers', async () => {
+    const renderResult = await render();
+    const { getByTestId } = renderResult;
+    expect(getByTestId('open-project-add-dialog-container')).toBeInTheDocument();
+  });
+
+  it('renders project to programming modal', async () => {
+    const renderResult = await render();
+    const { user, queryByTestId, getByTestId } = renderResult;
+
+    expect(getByTestId('open-project-programmed-dialog')).toBeInTheDocument();
```

Comment:
Use findByTestId and store the result into constant to query from it input fields. This will speedup the tests